sample/src/main/java/io/pixsight/sample/scenemanager/SampleActivity.kt

```diff
@@ -21,50 +21,67 @@ import kotlinx.android.synthetic.main.spinner.*
-        R.id.sample_switch_to_existing_layout ->
-            SampleNoAnnotationsActivity.startActivity(this)
-        R.id.sample_hide -> {
-            SceneManager.hide(this) // all is now hidden
-            SceneManager.scene(this, SAMPLE_HIDDEN) // load scene with a button so you can come back
+    private fun bindListeners() {
+        sample_switch_to_main?.setOnClickListener(this)
+        sample_switch_to_progress?.setOnClickListener(this)
+        go_to_main_from_loader?.setOnClickListener(this)
+        sample_switch_to_placeholder?.setOnClickListener(this)
+        go_to_main_from_placeholder?.setOnClickListener(this)
+        sample_switch_to_custom_view?.setOnClickListener(this)
+        sample_switch_to_activity?.setOnClickListener(this)
+        sample_switch_to_fragment?.setOnClickListener(this)
+        sample_switch_to_fragment_v4?.setOnClickListener(this)
+        sample_switch_to_existing_layout?.setOnClickListener(this)
+        sample_switch_to_bind_scenes?.setOnClickListener(this)
+        sample_switch_to_iod?.setOnClickListener(this)
+        sample_hide?.setOnClickListener(this)
+        restore_from_hidden_bt?.setOnClickListener(this)
+    }
+
+    override fun onClick(v: View) {
+        when (v.id) {
+            R.id.sample_switch_to_main,
+            R.id.sample_switch_to_activity,
+            R.id.go_to_main_from_loader,
+            R.id.restore_from_hidden_bt,
+            R.id.go_to_main_from_placeholder ->
+                SceneManager.scene(this, Scene.MAIN)
+            R.id.sample_switch_to_progress -> SceneManager.scene(this, Scene.SPINNER)
+            R.id.sample_switch_to_placeholder -> SceneManager.scene(this, Scene.PLACEHOLDER)
+            R.id.sample_switch_to_custom_view -> SceneManager.scene(this, SAMPLE_WITH_VIEW)
+            R.id.sample_switch_to_fragment ->
+                @Suppress("DEPRECATION")
+                SampleFragment.newInstance().show(fragmentManager, "SampleFragment")
+            R.id.sample_switch_to_fragment_v4 ->
+                SampleFragmentV4.newInstance().show(supportFragmentManager, "SampleFragmentv4")
+            R.id.sample_switch_to_existing_layout ->
+                SampleNoAnnotationsActivity.startActivity(this)
+            R.id.sample_switch_to_bind_scenes ->
+                SampleBindScenesActivity.startActivity(this)
+            R.id.sample_switch_to_iod ->
+                SampleInflateOnDemandActivity.startActivity(this)
+            R.id.sample_hide -> {
+                SceneManager.scene(this, SAMPLE_HIDDEN, true) // load a scene with a button so you can come back
+                SceneManager.hide(this, true) // all is now hidden
+                SceneManager.restore(this, true) // restore to SAMPLE_HIDDEN
+            }
+            else -> throw IllegalArgumentException("Nope")
         }
-        else -> throw IllegalArgumentException("Nope")
+
+        bindListeners()
```

Comment:
Listeners need to be rebind on every button clicked?

Reply:
Les vues n'existent pas avant le inflate on demand. C'est pourquoi je rebind.